x-pack/legacy/plugins/es_task_management/public/app/services/http/use_request.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO
+// This code was largely borrowed from request.ts in es_ui_shared
+// It was modified to use the http service from NP
+// Once all apps utilizing request.ts have been migrated to NP,
+// request.ts should be updated and this file will no longer be needed.
+
+import { useEffect, useState, useRef } from 'react';
+
+import { httpService } from './index';
+
+export interface SendRequestResponse {
+  data: any;
+  error: Error | null;
+}
+
+export interface SendRequestOptions {
+  method: 'get' | 'post' | 'put' | 'delete' | 'patch' | 'head';
+  body?: any;
+}
+
+export interface UseRequestConfig {
+  path: string;
+  requestOptions: SendRequestOptions;
+  pollIntervalMs?: number;
+  initialData?: any;
+  deserializer?: (data: any) => any;
+}
+
+export interface UseRequestResponse {
+  isInitialRequest: boolean;
+  isLoading: boolean;
+  error: null | unknown;
+  data: any;
+  sendRequest: (...args: any[]) => Promise<SendRequestResponse>;
+}
+
+export const sendRequest = async (
```

Comment:
I left a comment at the top of the file. Since I'm using the NP http service, I would have to modify the `request` lib in `es_ui_shared` in order for it to work. Once we migrate any apps using `request.ts` to use the NP http service, we can update the file directly and remove this instance here.